### PR TITLE
Visualisation in clustering

### PR DESCRIPTION
**Pull request by MVolobueva**
Add function for builing histogram visualization of cluster size disrtibution

### CODE REVIEW

File: vignettes/web_only/BCRpipeline.Rmd [hunk 1]
```diff
@@ -138,7 +139,7 @@ If  `.ref_only_first = FALSE`, you will get several different germlines:
 ```{r example 5}
 bcrdata$data %>%
     top(1) %>%
-    repGermline(.ref_only_first = FALSE, .threads = 1) %>% .$full_clones %>% .$Germline.sequence
+    repGermline(.ref_only_first = F, .threads = 1) %>% .$full_clones %>% .$Germline.sequence
```
Comment (Alexander230) [inline vignettes/web_only/BCRpipeline.Rmd:142]:
```suggestion
    repGermline(.ref_only_first = FALSE, .threads = 1) %>% .$full_clones %>% .$Germline.sequence
```

File: vignettes/web_only/BCRpipeline.Rmd [hunk 2]
```diff
@@ -188,6 +189,16 @@ Requirements for the input table for `repAlignLineage()`
...
 - must contain the 'Sequence.germline' column generated by repGermline() function.
 
+Align sequences in a cluster can be visualized using standard functions:
+
+```{r align visualisation}
+# A name of the first cluster
+align_dt$full_clones$Cluster[[1]]
+
+# Alignment of sequences from the first cluster
+image(align_dt$full_clones$Alignment[[1]], grid = T)
```
Comment (Alexander230) [inline vignettes/web_only/BCRpipeline.Rmd:199]:
```suggestion
image(align_dt$full_clones$Alignment[[1]], grid = TRUE)
```

File: vignettes/web_only/clustering.Rmd [hunk 3]
```diff
@@ -140,9 +140,12 @@ In general, the number of clusters you get depends on threshold you defined prev
 
 Requires defining the minimum percentage of similarity for the sequences in your cluster in advance. For example, if you want to find cluster of BCRs that not only have the same [V.name](http://V.name), [J.name](http://J.name), and CDR3 length — but also have more than 90 percent match in their CDR3.
 
-```{r example 8}
+```{r example 8, warning=FALSE}
```
Comment (Alexander230) [inline vignettes/web_only/clustering.Rmd:143]:
```suggestion
```{r example 8, warning = FALSE}
```

File: vignettes/web_only/clustering.Rmd [hunk 4]
```diff
@@ -152,9 +155,12 @@ clustTCR$"A2-i129" %>% .$Cluster %>% unique() %>% length()
 
 Requires directly determining the number of mismatched nucleotides. In case of TCR-recognising the same epitope, use 1 amino acid mismatch in CDR3 sequences.
 
-```{r example 9}
+```{r example 9, warning=FALSE}
```
Comment (Alexander230) [inline vignettes/web_only/clustering.Rmd:158]:
```suggestion
```{r example 9, warning = FALSE}
```

File: vignettes/web_only/clustering.Rmd [hunk 5]
```diff
@@ -164,9 +170,12 @@ clustTCR$"A2-i129" %>% .$Cluster %>% unique() %>% length()
 
 If you want to guarantee that the matches in the sequences are at a set proximity, specify a distance threshold 
 
-```{r example 10}
+```{r example 10, warning=FALSE}
```
Comment (Alexander230) [inline vignettes/web_only/clustering.Rmd:173]:
```suggestion
```{r example 10, warning = FALSE}
```

File: vignettes/web_only/clustering.Rmd [hunk 6]
```diff
@@ -176,6 +185,24 @@ clustTCR$"A2-i129" %>% .$Cluster %>% unique() %>% length()
 
 If you made it to this part, then you have successfully clustered your sequences. Congratulations! Now you have a new column `“cluster”` in your data sheet. You can now generalize the concept of clonotype based on sequence similarity and consider clusters as clonotypes.
 
+Comparing cluster size distributions is a good way to analyse quality of clustering process and to choose the best parameters for functions:
+
+```{r }
+#write small function for the visualization of cluster size destribution
+mk_hist <- function(clust_dt, graph_name){
```
Comment (Alexander230) [inline vignettes/web_only/clustering.Rmd:192]:
```suggestion
mk_hist <- function(clust_dt, graph_name) {
```

File: vignettes/web_only/clustering.Rmd [hunk 7]
```diff
@@ -176,6 +185,24 @@ clustTCR$"A2-i129" %>% .$Cluster %>% unique() %>% length()
...
+  return(ggplot(clust_dt, aes(x=cluster_size)) + geom_histogram( binwidth=1, fill="#69b3a2", color="#e9ecef", alpha=0.9) +
+           ggtitle(graph_name) +
+           theme_bw() +
+           theme(
+             plot.title = element_text(size=15)
+           ))
+}
+
+#compare plots
```
Comment (Alexander230) [inline vignettes/web_only/clustering.Rmd:201]:
```suggestion
# compare plots
```

File: vignettes/web_only/clustering.Rmd [hunk 8]
```diff
@@ -176,6 +185,24 @@ clustTCR$"A2-i129" %>% .$Cluster %>% unique() %>% length()
...
+           ggtitle(graph_name) +
+           theme_bw() +
+           theme(
+             plot.title = element_text(size=15)
+           ))
+}
+
+# compare plots
+mk_hist(dt_perc_similarity, '.perc_similarity=0.75') + mk_hist(dt_fixed_threshold, '.fixed_threshold=2') +
+  mk_hist(dt_nt_similarity, '.dt_nt_similarity=10')
```
Comment (MVolobueva) [inline vignettes/web_only/clustering.Rmd:203]:
```suggestion
  mk_hist(dt_nt_similarity, '.nt_similarity=10')
```

File: vignettes/web_only/BCRpipeline.Rmd [hunk 9]
```diff
@@ -128,6 +127,8 @@ Available species: "HomoSapiens" (default), "MusMusculus", "BosTaurus", "Camelus
 
 `.min_nuc_outside_cdr3` — filters out reads with low numbers of nucleotides from V or J gene outside of CDR3 region
```
Comment (VolkovaUE) [inline vignettes/web_only/BCRpipeline.Rmd:128]: filters out the readings with low nucleotide numbers

**Review by VolkovaUE — APPROVED** [review]

File: vignettes/web_only/BCRpipeline.Rmd (hunk 9 above)
Comment (Alexander230) [reply vignettes/web_only/BCRpipeline.Rmd:128]: I noticed that we copied this description from older version of germline.R, and in new version there is more clear description of this parameter: `this parameter sets how many nucleotides should have V or J chain outside of CDR3 to be considered good for further alignment`

Comment (Alexander230) [reply vignettes/web_only/BCRpipeline.Rmd:128]: We can also add that "reads with too short chains are filtered out"

Comment (MVolobueva) [inline vignettes/web_only/BCRpipeline.Rmd:128]:
```suggestion
`.min_nuc_outside_cdr3` — this parameter sets how many nucleotides should have V or J chain outside of CDR3 to be considered good for further alignment. Reads with too short chains are filtered out
```

**Pull request merged**